app/channel_test.go

```diff
@@ -104,3 +109,164 @@ func TestMoveChannel(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestJoinDefaultChannelsTownSquare(t *testing.T) {
+	th := Setup().InitBasic()
+	defer th.TearDown()
+
+	// figure out the initial number of users in town square
+	townSquareChannelId := store.Must(th.App.Srv.Store.Channel().GetByName(th.BasicTeam.Id, "town-square", true)).(*model.Channel).Id
+	initialNumTownSquareUsers := len(store.Must(th.App.Srv.Store.ChannelMemberHistory().GetUsersInChannelDuring(model.GetMillis()-100, model.GetMillis()+100, townSquareChannelId)).([]*model.ChannelMemberHistory))
+
+	// create a new user that joins the default channels
+	user := th.CreateUser()
+	th.App.JoinDefaultChannels(th.BasicTeam.Id, user, model.ROLE_CHANNEL_USER.Id, "")
```

Comment:
`model.ROLE_CHANNEL_USER` doesn't exist anymore. Use `model.CHANNEL_USER_ROLE_ID`.

Reply:
Noted and fixed when I rebased this morning.